### ISSUE TITLE
Is the WithdrawExpired message really required?

### ISSUE THREAD

**Issue by andrevmatos** [raiden-network/raiden#4721]
## Question

I'm implementing the withdraw handling functionality on the Light Client, and came across the [withdraw workflow](https://github.com/raiden-network/raiden/blob/develop/docs/adr/0005-withdraw-expiry.rst).
Question: why there exists the `WithdrawExpired` message? I see the point of `LockExpired` is to sync with partner an up-to-date balance proof for our side of the channel without the now useless lock, on top of which new balance proofs can be added (e.g. new transfers) without clogging the pending locks list with locks which can't be unlocked anyway. But withdraw doesn't carry a balance proof, so both sides could clear the pending withdraw lists on their own e.g. confirmation_blocks after expiration, without the need to sync this through messages.

Example based on simple example from docs:
- confirmation_blocks = 5, current_block = 123 , withdraw expiration timeout = 20 blocks
- Alice's balance = 100 tokens
- Alice requests a withdraw of 40 tokens expiring at block=123+20=143. In this case, usable balance of Alice is 60 (assuming it did not send/receive transfers to/from Bob).
- If Bob sees the `WithdrawRequest`, he also sets Alice's `total_withdraw=40`, responds with a `WithdrawConfirmation`, and Alice sends the on-chain tx which is mined before `block=143`, Bob and Alice sees the `ChannelWithdraw` event and settle Alice's `total_withdraw=40`
- But if Bob is offline or not cooperating, or Alice for any reason decides or can't send the on-chain tx (e.g. out of gas), during the expiration period both users still have Alice's `total_withdraw=40`, as the withdraw tx might still go through.
- After `expiration_block+confirmation=143+5=148`, both Alice and Bob can be sure withdraw tx can't be performed on-chain anymore (independently of the WithdrawConfirmation message having been sent/received or not), so they can simply clear the pending withdraw request (even e.g. if Bob comes back online much later). Both Bob and Alice see Alice's `total_withdraw=0`

One less message to handle. No change on smartcontract side.

Is there any specific reason for the `WithdrawExpired` roundtrip I may be missing?

**Comment by andrevmatos** [comment]
If there's any concern related to Alice using the unlocked balance while Bob still didn't the `block=148`, as this value only limits new transfers sent signed by Alice, she could play safe and expire her withdraw only e.g. `2 * confirmation_blocks`, `block=143+2*5=153` , to give "time" to Bob to catch up without risk of they getting out-of-sync, or just rely on `LockedTransfer` retry, as eventually he'd clear the pending withdraw request and accept the retried `LockedTransfer` which makes use of the [un]locked amount.

**Comment by konradkonrad** [comment]
I tend to agree. The `WithdrawExpired` message really caters only one (quite edgy) case. For Alice requesting a cooperative withdraw with Bob, it only helps if Bob is uncooperative about the withdraw but otherwise stays completely canonical. So Bob would need to ignore the `WithdrawRequest` but then keep accepting all further protocol interactions. 

For this particular case `WithdrawExpired` can help to "clarify" the state of the channel around the block of expiry.

The proposal of implicit expiry may lead to some extra latency around that particular block, because a transfer that relies on the withdraw-locked balance could be understood as invalid from Bob's point of view, if they are lagging in synchronization.

**Comment by andrevmatos** [comment]
> The proposal of implicit expiry may lead to some extra latency around that particular block, because a transfer that relies on the withdraw-locked balance could be understood as invalid from Bob's point of view, if they are lagging in synchronization.

I agree, but since this isn't expected to happen, and if the withdrawer takes care of giving a little bit more time before considering the funds unlocked (e.g. by waiting 2 * confirmationBlocks), this shouldn't happen, and anyway, would only _hurt_ liquidity for at most a couple of blocks, so no big deal either.

Only thing we need to be careful is **when** we should consider the amount as unlocked in order to calculate imbalance fees (when they're enabled again), and have that replicated on PFS. Inconsistencies here may led to transfers failing due to wrong fees, but I wouldn't expect this to happen, since this only reduce the capacity on one side, and not add on the other while the funds are locked.
